### ISSUE TITLE
Appetite for updating dependencies/Build System?

### ISSUE THREAD

**Issue by playmer** [Galladite27/ONScripter-EN#4]
Hi there, I have some interest in utilizing this fork for some enhancements in an old nscripter game. I've worked on my own fork of onscripter for this quite some time ago, but I'd prefer to contribute changes to a well maintained fork than leave them to rot in my own.

The primary things I worked on were updating the various dependencies to latest (albeit a few were replaced), getting resize support working (game maintains aspect ratio, but will resize to the borders of the window where possible), controller support. ,and CMake so it could be relatively easily built on Window/Mac/Linux, I also built in a very goofy system for texture replacement, but I wouldn't really want to replicate that.

When I initially did it, I cut out a lot of support for the more esoteric platforms. In working on it in my fork of this repo, it's likely I'm breaking them, but I'm not going out of my way to delete them. But I don't have working toolchains for most of them. Just looking around the code, I think the only one that might be impossible to port to (and of course, I could certainly be wrong) is the PDA.

Most of the upgrades have gone smoothly so far, my last large hurdle are the various media file wrappers as well as CD support. I'm looking at possibly replacing the former with a wrapper for FFMpeg, and the latter with turning the old SDL1 CD module into a tiny library of it's own and see I can get it working that way.

All that being said, before I make a relatively large PR whenever I finish this, would there be interest in merging such a thing? If need be I suppose I can keep a fork going and try to keep up to date with the upstream, I don't have any desire to compete so it would just be used for the few projects I might help out with.

**Comment by Galladite27** [comment]
Hi, thanks for asking. We're happy to continue using the current build system, however some of the other features such as window resizing and CD support would be greatly appreciated. If you get in contact once they're done, I'll be happy to implement your changes myself to this branch, and give you credit. If you ever want any support regarding the changes made since the start of this continuation, I'll be happy to help.

Just to clarify, you've been making these changes so far to an old version of ONScripter-EN, right? So are you planning to re-implement all your fixes on top on this new version?

**Comment by playmer** [comment]
Yeah, I've already started over here: https://github.com/playmer/ONScripter-EN-Official/tree/CMakeAndDepUpgrade

So far I've updated all the deps, but the mpeg/movie commands aren't fully done. I rewrote AviWrapper into FFMpegWrapper so in theory the Avi command should work, but I don't have a way to test any of those. (If you're aware of a way to test them let me know!)

I'm not sure you'll be able to implement the resize stuff in the same way without also updating the dependencies, the big one is moving to SDL2 (and eventually SDL3, but that's not ready yet). I think it's likely there's a bunch of code that can be removed with these updates, such as the in-built ogg decoder, MadWrapper, use of smpeg for mp3 decoding. Pretty sure new versions of sdl_mixer have all that stuff built in. For now though I've maintained those. 

Resizing is roughly half done in that branch, I need to rescale mouse input to the new screen. It uses windowed full screen now. 

Basically it's not ready for real use yet (haven't even spent time to get it compiling on Linux/Mac/Mingw again yet), but I'm getting it almost back to where I'd like it. After I'm done with that, I know someone who wants the insertmenu and such implemented, so no promises, but I'll probably look at that. Tentative plan is to create a Qt window to have SDL render into and have a bridge between them to let those commands do their work.
